### PR TITLE
Rename `ObjectSchemaProperty` to `PropertySchema`

### LINKED ISSUE #5183: Rename `ObjectSchemaProperty`/`CanonicalObjectSchemaProperty` to `PropertySchema`/`CanonicalPropertySchema`
Renames the relaxed and canonical representations of a property's schema to a more suitable name and consistent with the other names such as `RealmSchema` and `ObjectSchema`.

### PR DESCRIPTION
**Pull request by elle-j**
## What, How & Why?

This closes #5183.

`CanonicalObjectSchemaProperty` and `ObjectSchemaProperty` are renamed to `CanonicalPropertySchema` and `PropertySchema` respectively.

This aligns with the existing names `RealmSchema` and `ObjectSchema`.

(The old names will still be exported as `export type ObjectSchemaProperty = PropertySchema` to prevent a breaking change. Complete removal of the old names are planned for v13.)

Files not updated (since they are planned to be removed):
* Files in `/docs`
* `/types/index.d.ts`

## ☑️ ToDos
* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [x] 🚦 Run tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated

### CODE REVIEW

File: packages/realm/src/schema/types.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export type RelationshipPropertyTypeName = "object" | "linkingObjects";
  */
 export type UserTypeName = string;
 
-// TODO: Can be removed when ObjectSchemaProperty is removed
+// TODO: Can be removed if/when integration tests no longer use it.
```
Comment (kneth) [inline packages/realm/src/schema/types.ts:68]: Please create an issue for it so we don't forget

**Review by kneth — COMMENTED** [review]
* Do we need to state the change in `CHANGELOG.md`?
* Should the API documentation be updated?
* We probably need to notify the docs team about this change (by creating a `DOCSP` ticket)

**Review by takameyer — APPROVED** [review]
I like this naming convention.  Excited to get this change live!

**Comment by kraenhansen** [comment]
As discussed in person, we might want to wait merging this until we've made the `v12` release.

**Comment by elle-j** [comment]
Update: The renaming will be merged as part of v12; however, in order to not cause a breaking change we decided to continue exporting the old name `export type ObjectSchemaProperty = PropertySchema;` with a deprecation label. (Planned to be removed v13.)

File: packages/realm/src/schema/types.ts (hunk 1 above)
Comment (elle-j) [reply packages/realm/src/schema/types.ts]: Update: This has now been resolved in this PR 👍

**Pull request merged**